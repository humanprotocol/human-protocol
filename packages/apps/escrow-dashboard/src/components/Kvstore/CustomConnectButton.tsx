import Button from '@mui/material/Button';
import Avatar from '@mui/material/Avatar';
import Box from '@mui/material/Box';
import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
import { ConnectButton } from '@rainbow-me/rainbowkit';
export const CustomConnectButton = () => {
  return (
    <ConnectButton.Custom>
      {({ account, chain, openAccountModal, openConnectModal, mounted }) => {
        // Note: If your app doesn't use authentication, you
        // can remove all 'authenticationStatus' checks
        const ready = mounted;
        const connected = ready && account && chain;
        return (
          <div
            {...(!ready && {
              'aria-hidden': true,
              style: {
                opacity: 0,
                pointerEvents: 'none',
                userSelect: 'none',
              },
            })}
          >
            {(() => {
              if (!connected) {
                return (
                  <Button
                    sx={{ marginLeft: { xs: 0.7, sm: 0.7, md: 0 } }}
                    onClick={openConnectModal}
                    variant="outlined"
                  >
                    Connect Wallet
                  </Button>
                );
              }
              if (chain.unsupported) {
                return (
                  <Button onClick={openConnectModal} variant="outlined">
                    Wrong Network
                  </Button>
                );
              }
              return (
                <Box sx={{ display: 'flex', gap: 12 }}>
                  <Button
                    endIcon={<ArrowDropDownIcon />}
                    onClick={openAccountModal}
                    variant="contained"
                    sx={{
                      '&:hover': {
                        color: 'gray',
                        backgroundColor: '#f6f7fd',
                      },
                      borderRadius: 12,
                      paddingLeft: 1,
                      paddingTop: 0.4,
                      paddingBottom: 0.4,
                      backgroundColor: '#f6f7fe',
                      color: '#320a8d',
                    }}
                    startIcon={
                      <Avatar
                        src={
                          'https://cdn.stamp.fyi/avatar/eth:' +
                          account.displayName
                        }
                      />
                    }
                  >
                    {' '}
                    {account.displayName}
                  </Button>
                </Box>
              );
            })()}
          </div>
        );
      }}
    </ConnectButton.Custom>
  );
};
