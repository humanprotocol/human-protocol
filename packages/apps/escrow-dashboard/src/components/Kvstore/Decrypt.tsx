import KVStore from '@human-protocol/core/abis/KVStore.json';
import { ChainId, NETWORKS } from '@human-protocol/sdk';
import { Close as CloseIcon } from '@mui/icons-material';
import {
  Box,
  Button,
  Divider,
  Grid,
  Paper,
  Snackbar,
  TextField,
  Typography,
} from '@mui/material';
import * as openpgp from 'openpgp';
import { useState, ChangeEvent, FC } from 'react';
import { useContractRead, useAccount, useNetwork } from 'wagmi';

import { Alert } from '../Alert';
import { showIPFS } from 'src/services';

export const Decrypt: FC = () => {
  const { chain } = useNetwork();
  const [key, setKey] = useState('');
  const [passphrase, setPassphrase] = useState('');
  const [value, setValue] = useState('');
  const [error, setError] = useState('');
  const [privkey, setPrivkey] = useState('');
  const [decrypted, setDecrypted] = useState('');
  const [copy, setCopy] = useState<boolean>(false);
  const [loading, setLoading] = useState<boolean>(false);
  const [filename, setFilename] = useState<string>('');

  const { address } = useAccount();
  const { refetch } = useContractRead({
    address: NETWORKS[chain?.id as ChainId]?.kvstoreAddress as `0x${string}`,
    abi: KVStore,
    functionName: 'get',
    args: [address, key],
  });

  async function getValue() {
    setLoading(true);
    try {
      const { data } = await refetch();
      setValue(await showIPFS(data!.toString()));
      setLoading(false);
    } catch (e) {
      if (e instanceof Error) {
        setLoading(false);
        setError(e.message);
      }
    }
  }

  const handleFile = async (e: ProgressEvent<FileReader>) => {
    setLoading(true);
    setPrivkey('');

    try {
      const content = e.target!.result as string;
      const a = await openpgp.readKey({ armoredKey: content });
      if (!a.isPrivate()) {
        setError('Error, file is not private key');
        setLoading(false);
        setFilename('');
        return;
      }
      setPrivkey(content);
      setLoading(false);
    } catch (e) {
      if (e instanceof Error) {
        setError(e.message);
        setLoading(false);
        setFilename('');
      }
    }
  };

  const handleChange = (ev: ChangeEvent<HTMLInputElement>) => {
    const reader = new FileReader();
    reader.addEventListener('load', handleFile);
    reader.readAsText(ev.target.files![0]);
    setFilename(ev.target.files![0].name);
  };

  async function decryptValue() {
    setLoading(true);
    try {
      const privateKey = await openpgp.decryptKey({
        privateKey: await openpgp.readPrivateKey({ armoredKey: privkey }),
        passphrase,
      });

      const message = await openpgp.readMessage({
        armoredMessage: value,
      });

      const { data: decryptedVal } = await openpgp.decrypt({
        message,
        decryptionKeys: privateKey,
      });

      setDecrypted(decryptedVal as string);
      setLoading(false);
    } catch (e) {
      if (e instanceof Error) {
        setDecrypted('');
        setError(e.message);
        setLoading(false);
      }
    }
  }
  return (
    <Grid
      item
      xs={12}
      sm={12}
      md={12}
      container
      direction="column"
      justifyContent="center"
      alignItems="center"
    >
      <Snackbar
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        open={copy}
        autoHideDuration={3000}
        onClose={() => setCopy(false)}
      >
        <Alert
          onClose={() => setCopy(false)}
          severity="info"
          sx={{ width: '100%' }}
        >
          copied
        </Alert>
      </Snackbar>

      <Snackbar
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        open={error.length > 0}
        autoHideDuration={3000}
        onClose={() => setError('')}
      >
        <Alert
          onClose={() => setError('')}
          severity="error"
          sx={{ width: '100%' }}
        >
          {error}
        </Alert>
      </Snackbar>
      <Box sx={{ width: { xs: 1, md: '50%', lg: '50%', xl: '40%' } }}>
        <Paper>
          <Box sx={{ borderBottom: '1px solid #CBCFE6' }}>
            <Grid
              container
              direction="row"
              justifyContent={'space-between'}
              flexDirection={{ xs: 'column-reverse', md: 'row' }}
            >
              <Grid item>
                <Typography fontWeight={`500`} padding={`10px`}>
                  Decrypt
                </Typography>
              </Grid>
            </Grid>
          </Box>
          <Grid
            container
            direction="column"
            sx={{
              padding: {
                xs: '0px 80px 0px 80px',
                md: '0px 80px 0px 80px',
                lg: '0px 80px 0px 80px',
                xl: '0px 80px 0px 80px',
              },
            }}
          >
            <Grid
              item
              container
              direction="column"
              sx={{
                marginBottom: { lg: 10 },
              }}
            >
              <Box sx={{ mt: { md: 10, lg: 10, xl: 10 }, mb: 2 }}>
                <Typography>
                  Decrypt using your public key and private key
                </Typography>
              </Box>

              <TextField
                sx={{ my: 1 }}
                id="outlined-basic"
                label="Key"
                variant="outlined"
                value={key}
                onChange={(e) => setKey(e.target.value)}
                disabled={loading}
              />
              <Box>
                <Button
                  disabled={loading}
                  onClick={getValue}
                  sx={{ my: 1, mb: !value ? 5 : 2 }}
                  variant="outlined"
                >
                  Get
                </Button>
              </Box>
              {value && (
                <>
                  <Box>
                    <Typography fontWeight={500}>Value</Typography>
                  </Box>
                  <Box sx={{ width: { xs: 1 } }}>
                    <Box
                      className="pubkey"
                      sx={{
                        backgroundColor: '#f6f7fe',
                        maxHeight: 200,
                        marginTop: 2,
                        overflowY: 'scroll',
                        overflowWrap: 'break-word',
                        padding: 4,
                        borderRadius: 3,
                      }}
                    >
                      {value}
                    </Box>
                  </Box>
                  <Box sx={{ marginBottom: { xs: 2, sm: 2, md: 2, lg: 0 } }}>
                    <Button
                      onClick={() => {
                        setCopy(true);
                        navigator.clipboard.writeText(value);
                      }}
                      size="small"
                    >
                      Copy
                    </Button>
                  </Box>
                  <Divider sx={{ my: 2 }} light />
                  <Box>
                    <Typography>Decrypt </Typography>
                  </Box>
                  <Box
                    display={`flex`}
                    flexDirection={'row'}
                    alignContent={`center`}
                  >
                    {privkey.length === 0 && (
                      <Button
                        sx={{ my: 2 }}
                        disabled={loading}
                        variant="outlined"
                        size="small"
                        component="label"
                      >
                        Upload Private Key
                        <input
                          type="file"
                          aria-label="add files"
                          hidden
                          onChange={handleChange}
                        />
                      </Button>
                    )}
                    {privkey.length > 0 && (
                      <>
                        <Typography sx={{ my: 2 }} variant={`body1`}>
                          {filename}
                        </Typography>
                        <CloseIcon
                          onClick={() => setPrivkey('')}
                          sx={{ cursor: 'pointer', my: 2.3 }}
                        />
                      </>
                    )}
                  </Box>
                  <TextField
                    sx={{ my: 1 }}
                    id="outlined-basic"
                    label="Passphrase"
                    variant="outlined"
                    value={passphrase}
                    type="password"
                    disabled={loading}
                    onChange={(e) => setPassphrase(e.target.value)}
                  />
                  <Box
                    display={`flex`}
                    width={`100%`}
                    justifyContent={`flex-end`}
                  >
                    <Button
                      disabled={loading}
                      sx={{ my: 1, mb: !decrypted ? 5 : 2 }}
                      variant="contained"
                      onClick={decryptValue}
                    >
                      Decrypt
                    </Button>
                  </Box>
                  {decrypted && (
                    <>
                      <Box>
                        <Typography fontWeight={`500`}>
                          Decrypted Value{' '}
                        </Typography>
                      </Box>
                      <Box
                        className="pubkey"
                        sx={{
                          backgroundColor: '#f6f7fe',
                          maxHeight: 200,
                          marginTop: 2,
                          overflowY: 'scroll',
                          overflowWrap: 'break-word',
                          padding: 4,
                          borderRadius: 3,
                        }}
                      >
                        {decrypted}
                      </Box>
                      <Box
                        sx={{ marginBottom: { xs: 2, sm: 2, md: 2, lg: 0 } }}
                      >
                        <Button
                          onClick={() => {
                            setCopy(true);
                            navigator.clipboard.writeText(value);
                          }}
                          size="small"
                        >
                          Copy
                        </Button>
                      </Box>
                    </>
                  )}
                </>
              )}
            </Grid>
          </Grid>
        </Paper>
      </Box>
    </Grid>
  );
};
