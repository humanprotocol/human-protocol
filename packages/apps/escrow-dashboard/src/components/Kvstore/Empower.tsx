import { Grid, Paper, Typography, Box, Button } from '@mui/material';
import DoneIcon from '@mui/icons-material/Done';
import React, { Dispatch } from 'react';

export const Empower = ({
  refetch,
  setPublicKey,
}: {
  refetch: any;
  setPublicKey: Dispatch<string>;
}): React.ReactElement => {
  async function finish() {
    try {
      const { data } = await refetch();
      setPublicKey(data);
    } catch (e) {
      if (e instanceof Error) {
      }
    }
  }
  return (
    <Grid container>
      <Grid item container direction="row" justifyContent="center">
        <Grid
          item
          xs={12}
          sm={12}
          md={12}
          container
          direction="column"
          justifyContent="center"
          alignItems="center"
        >
          <Box>
            <Paper>
              <Grid container direction="column">
                <Grid
                  item
                  container
                  direction="column"
                  alignItems="center"
                  sx={{
                    marginTop: { xs: 1, sm: 1, md: 10, lg: 10 },
                    paddingLeft: 10,
                    paddingRight: 10,
                    marginBottom: { md: 10, lg: 10 },
                  }}
                >
                  <DoneIcon />
                  <Typography
                    sx={{ marginTop: 3 }}
                    variant="h6"
                    color="primary"
                  >
                    Success!
                  </Typography>
                  <Typography
                    sx={{ marginTop: 1 }}
                    variant="body2"
                    color="primary"
                  >
                    Your public key has been stored on the blockchain now, click
                    on continue to encrypt or decrypt
                  </Typography>
                  <Button
                    onClick={finish}
                    variant="contained"
                    sx={{ marginTop: 2 }}
                  >
                    Continue
                  </Button>
                </Grid>
              </Grid>
            </Paper>
          </Box>
        </Grid>
      </Grid>
    </Grid>
  );
};
