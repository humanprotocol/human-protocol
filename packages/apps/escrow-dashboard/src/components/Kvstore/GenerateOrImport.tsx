import { Grid, Paper, Typography, Button } from '@mui/material';
import { Dispatch, FC } from 'react';

import keySvg from 'src/assets/key.svg';

export type GenerateOrImportProps = {
  setStep: Dispatch<number>;
  setPage: Dispatch<number>;
};

export const GenerateOrImport: FC<GenerateOrImportProps> = ({
  setStep,
  setPage,
}) => (
  <Paper>
    <Grid container justifyContent="center" direction="column">
      <Grid
        item
        container
        direction="column"
        alignItems="center"
        sx={{
          marginTop: { xs: 1, sm: 1, md: 10, lg: 10 },
          marginBottom: { lg: 10 },
        }}
      >
        {' '}
        <img width="100" src={keySvg} alt="lbank" />{' '}
        <Typography sx={{ marginTop: 3 }} variant="body2" color="primary">
          If you already have your public key import it, if not generate it
        </Typography>
      </Grid>
      <Grid
        item
        container
        direction="row"
        justifyContent="center"
        alignItems="flex-end"
        sx={{
          marginTop: { xs: 1, sm: 1, md: 10, lg: 10 },

          marginBottom: { xs: 1, sm: 1, md: 10, lg: 7 },
        }}
      >
        <Button
          onClick={() => {
            setStep(1);
            setPage(1);
          }}
          variant="contained"
          sx={{ marginRight: 2 }}
        >
          Generate
        </Button>
        <Button
          onClick={() => {
            setStep(1);
            setPage(2);
          }}
          variant="outlined"
        >
          Import
        </Button>
      </Grid>
    </Grid>
  </Paper>
);
