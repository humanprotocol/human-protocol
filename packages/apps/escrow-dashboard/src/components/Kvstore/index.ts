export { KvstoreView } from './KvstoreView';
export type Key = {
  publicKey: string;
  privateKey: string;
};
