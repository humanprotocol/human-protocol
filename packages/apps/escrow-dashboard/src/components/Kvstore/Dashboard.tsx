import { Grid, Box, Tabs, Tab } from '@mui/material';
import React, { useState, useEffect, Dispatch } from 'react';
import ViewTitle from '../ViewTitle';

import { StoredPubkey } from './StoredPubkey';
import { Decrypt } from './Decrypt';
import { Encrypt } from './Encrypt';
import { showIPFS } from '../../services/index';

export const Dashboard = ({
  publicKey,
  refetch,
  setPublicKey,
  setStep,
  setPage,
}: {
  publicKey: string;
  refetch: any;
  setPublicKey: Dispatch<string>;
  setStep: Dispatch<number>;
  setPage: Dispatch<number>;
}): React.ReactElement => {
  const [value, setValue] = React.useState(0);
  const [pubkey, setPubkey] = useState<string>('');
  useEffect(() => {
    if (publicKey.trim().length !== 0) {
      showIPFS(publicKey).then((a) => setPubkey(a));
    }
  }, [publicKey]);
  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setValue(newValue);
  };
  return (
    <Grid container>
      <Grid
        item
        xs={12}
        sm={12}
        md={12}
        direction="row"
        justifyContent="center"
      >
        <Box
          marginBottom="1em"
          display="flex"
          alignItems="center"
          flexWrap="wrap"
        >
          <ViewTitle title="ETH KV Store" iconUrl="/images/small_key.svg" />
        </Box>
        <Box sx={{ borderBottom: 1, borderColor: 'divider', marginBottom: 2 }}>
          {pubkey.trim().length > 0 && (
            <Tabs value={value} onChange={handleChange}>
              <Tab label="Public Key" />
              <Tab label="Encrypt" />
              <Tab label="Decrypt" />
            </Tabs>
          )}
        </Box>
        {value === 0 && (
          <StoredPubkey
            refetch={refetch}
            publicKey={pubkey}
            setStep={setStep}
            setPage={setPage}
            setPublicKey={setPublicKey}
          />
        )}
        {value === 1 && <Encrypt publicKey={pubkey} />}
        {value === 2 && <Decrypt />}
      </Grid>
    </Grid>
  );
};
