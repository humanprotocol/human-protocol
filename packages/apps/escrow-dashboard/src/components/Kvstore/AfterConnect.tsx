import {
  Box,
  Grid,
  Paper,
  Step,
  Stepper,
  StepLabel,
  Typography,
} from '@mui/material';
import { useState, Dispatch, FC } from 'react';

import { STEPS } from './constants';
import { Empower } from './Empower';
import { GenerateOrImport } from './GenerateOrImport';
import { GeneratePubkey } from './GeneratePubkey';
import { ImportPubkey } from './ImportPubkey';
import { Success } from './Success';
import { Key } from './types';

type StepPage = {
  step: number;
  setStep: Dispatch<number>;
  page: number;
  setPage: Dispatch<number>;
};

export type AfterConnectProps = {
  pubkeyExist: boolean;
  refetch: any;
  setPublicKey: Dispatch<string>;
} & StepPage;

export const AfterConnect: FC<AfterConnectProps> = ({
  step,
  setStep,
  page,
  setPage,
  pubkeyExist,
  refetch,
  setPublicKey,
}) => {
  const [key, setKey] = useState<Key>({ publicKey: '', privateKey: '' });

  return (
    <Grid container>
      <Grid
        item
        xs={12}
        sm={12}
        md={12}
        container
        direction="row"
        justifyContent="center"
      >
        <Grid
          item
          xs={12}
          sm={12}
          md={12}
          container
          direction="column"
          justifyContent="center"
          alignItems="center"
        >
          <Box>
            <Typography variant="h4" sx={{ marginBottom: 2 }} color="primary">
              ETH KV Store
            </Typography>
            <Paper sx={{ padding: { md: 2 }, marginBottom: 2 }}>
              {' '}
              <Box sx={{ width: '100%' }}>
                <Stepper activeStep={step}>
                  {STEPS.map((step) => (
                    <Step key={step}>
                      <StepLabel>{step}</StepLabel>
                    </Step>
                  ))}
                </Stepper>
              </Box>
            </Paper>

            {page === 0 && (
              <GenerateOrImport setStep={setStep} setPage={setPage} />
            )}
            {page === 1 && (
              <GeneratePubkey
                refetch={refetch}
                setPublicKey={setPublicKey}
                pubkeyExist={pubkeyExist}
                setKey={setKey}
                setStep={setStep}
                setPage={setPage}
              />
            )}
            {page === 1.5 && (
              <Success
                what="generated"
                keys={key}
                setStep={setStep}
                setPage={setPage}
              />
            )}
            {page === 2 && (
              <ImportPubkey
                refetch={refetch}
                setPublicKey={setPublicKey}
                pubkeyExist={pubkeyExist}
                setKey={setKey}
                setStep={setStep}
                setPage={setPage}
              />
            )}
            {page === 2.5 && (
              <Success
                what="imported"
                keys={key}
                setStep={setStep}
                setPage={setPage}
              />
            )}
            {page === 3 && (
              <Empower refetch={refetch} setPublicKey={setPublicKey} />
            )}
          </Box>
        </Grid>
      </Grid>
    </Grid>
  );
};
