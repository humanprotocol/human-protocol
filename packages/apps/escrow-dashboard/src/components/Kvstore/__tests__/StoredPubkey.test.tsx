import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { StoredPubkey } from 'src/components/Kvstore/StoredPubkey';

import { RainbowKitProvider } from '@rainbow-me/rainbowkit';
import {
  Providers,
  setupClient,
  getSigners,
  testChains,
} from '../../../../tests/utils';
import { MockConnector } from '@wagmi/core/connectors/mock';

describe('when rendered StoredPubkey component', () => {
  it('should render `text` prop', async () => {
    const client = setupClient({
      connectors: [
        new MockConnector({
          options: {
            signer: getSigners()[0]!,
            // Turn on `failConnect` flag to simulate connect failure
          },
        }),
      ],
    });
    await act(async () => {
      render(<StoredPubkey />, {
        wrapper: ({ children }: { children: React.ReactNode }) => (
          <Providers client={client}>
            <RainbowKitProvider chains={testChains} modalSize="compact">
              {children}
            </RainbowKitProvider>
          </Providers>
        ),
      });
    });
    expect(screen.getByText(/Generate New Key/)).toBeInTheDocument();
  });
});

it('StoredPubkey component renders correctly, corresponds to the snapshot', () => {
  const client = setupClient({
    connectors: [
      new MockConnector({
        options: {
          signer: getSigners()[0]!,
          // Turn on `failConnect` flag to simulate connect failure
        },
      }),
    ],
  });
  const tree = renderer
    .create(
      <Providers client={client}>
        <RainbowKitProvider chains={testChains} modalSize="compact">
          <StoredPubkey />
        </RainbowKitProvider>
      </Providers>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
