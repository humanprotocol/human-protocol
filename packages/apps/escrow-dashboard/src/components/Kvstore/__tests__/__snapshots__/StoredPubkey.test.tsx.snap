// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StoredPubkey component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 css-wprbfx-MuiGrid-root"
>
  <div
    className="MuiBox-root css-9p5j7y"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
    >
      <div
        className="MuiBox-root css-ud8e2h"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-tfl2rr-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-1wjnbhc-MuiTypography-root"
            >
              Stored Public key:
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-vi45jv-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Import New Key
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-1t6yn7y-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Generate New Key
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-115thbp-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-sllrir-MuiGrid-root"
        >
          <div
            className="MuiBox-root css-11mn6qt"
          >
            <div
              className="pubkey MuiBox-root css-1oixsss"
            >
              <p
                className="MuiTypography-root MuiTypography-body2 MuiTypography-alignJustify css-1mawy8d-MuiTypography-root"
              />
            </div>
          </div>
          <div
            className="MuiBox-root css-1su85v3"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Copy
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Refresh
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
